src/rgw/rgw_sal_motr.cc

```diff
@@ -1257,8 +1257,49 @@ MotrObject::~MotrObject() {
 
 int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
 {
-  // TODO: implement
-  ldpp_dout(dpp, 20) <<__func__<< ": MotrObject::set_obj_attrs()" << dendl;
+  // TODO : Set tags for multipart objects
+  if (this->category == RGWObjCategory::MultiMeta)
+    return 0;
+
+  string bname, key;
+  if (target_obj) {
```

Comment:
We should create a separate function for the common code used in set and get attr functions

Reply:
I agree, quite a few lines are common in `set_obj_attrs()` & `get_obj_attrs()`
You should create a new function which will contain common code from these two functions. See @selvakumaar5496 if you can do it as part of this PR otherwise @cdeshmukh this refactoring can be planned for next sprint.

Reply:
Yeah agree. Still, lets address this separately only in next sprint @selvakumaar5496 , as this ticket is quite critical to fix as of now and dont want to expand test scope with this change.
thanks @jjxsg 
@sjain09 hope you are fine too

Reply:
I am okay.